addons/web_editor/static/src/components/media_dialog/search_media.js

```diff
@@ -3,19 +3,33 @@
 import { debounce } from '@web/core/utils/timing';
 import { useAutofocus } from '@web/core/utils/hooks';
 
-import { Component, onWillUnmount, xml } from "@odoo/owl";
+import { Component, onWillUnmount, xml, useEffect, useState } from "@odoo/owl";
 
 export class SearchMedia extends Component {
     setup() {
         useAutofocus();
-        this.search = debounce((ev) => this.props.search(ev.target.value), 1000);
+        this.debouncedSearch = debounce(this.props.search, 1000);
+
+        this.state = useState({
+            input: '',
+        });
+
         onWillUnmount(() => {
-            this.search.cancel();
+            this.debouncedSearch.cancel();
```

Comment:
pretty sure [useDebounce](https://github.com/odoo/odoo/blob/5be91c78c687cc2ac8fd3f9a94e4e8ef2b951efa/addons/web/static/src/core/utils/timing.js#L117) will do that for you

```js
this.search = useDebounce(this.props.search, 1000)
```